core-library/src/main/java/org/silverpeas/core/admin/space/SpaceInst.java

```diff
@@ -351,9 +326,11 @@ public int getNumComponentInst() {
    * @param componentInst component instance to be added
    */
   public void addComponentInst(ComponentInst componentInst) {
-    data.safeWrite((p, s, c) -> {
-      c.removeIf(i -> i.getId().equals(componentInst.getId()));
-      c.add(componentInst);
+    data.safeWrite(d -> {
+      final String cId = componentInst.getId();
+      final List<String> c = d.getComponentIds();
+      c.removeIf(cId::equals);
+      c.add(cId);
```

Comment:
Why removing the component instance id if it is already in the list? Shouldn't be added only if it isn't yet in the list?

Reply:
It is expected here that the old representation of a component instance is removed before again added.

Reply:
But why added it again if it is already in the list? Why not keeping it in the list and add it only if it isn't in the list?

Reply:
Ho yes, now that it is only a list of identifier about components, it should be managed like you said into your first comment.
Head on the handlebars :-D

Reply:
:-)